src/integrationTest/java/com/mongodb/hibernate/type/temporal/UnsupportedItems.java

```diff
@@ -0,0 +1,248 @@
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.type.temporal;
+
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import java.util.Collection;
+import org.hibernate.annotations.Struct;
+
+final class UnsupportedItems {
+
+    private UnsupportedItems() {}
+
+    @Entity
+    static class ItemWithId<T> {
+        @Id
+        T id;
+    }
+
+    @Entity
+    static class ItemWithFlattenedEmbeddableId<T> {
+        @Id
+        FlattenedEmbeddableWithBasicPersistentAttribute<T> id;
+    }
+
+    @Entity
+    static class ItemWithArrayPersistentAttribute<T> {
+        @Id
+        int id;
+
+        T[] v;
+    }
+
+    @Entity
+    static class ItemWithBasicPersistentAttribute<T> {
+        @Id
+        int id;
+
+        T v;
+    }
+
+    @Entity
+    static class ItemWithCollectionPersistentAttribute<T> {
+        @Id
+        int id;
+
+        Collection<T> v;
+    }
+
+    @Entity
+    static class ItemWithEmbeddableWithBasicPersistentAttribute<T> {
+        @Id
+        int id;
+
+        FlattenedEmbeddableWithBasicPersistentAttribute<T> v;
+    }
+
+    @Entity
+    static class ItemWithEmbeddableWithArrayPersistentAttribute<T> {
+        @Id
+        int id;
+
+        FlattenedEmbeddableWithArrayPersistentAttribute<T> v;
+    }
+
+    @Entity
+    static class ItemWithEmbeddableWithCollectionPersistentAttribute<T> {
+        @Id
+        int id;
+
+        FlattenedEmbeddableWithCollectionPersistentAttribute<T> v;
+    }
+
+    @Entity
+    static class ItemWithNestedEmbeddableWithBasicPersistentAttribute<T> {
+        @Id
+        int id;
+
+        FlattenedEmbeddableNestedWithBasicPersistentAttribute<T> v;
+    }
+
+    @Entity
+    static class ItemWithNestedEmbeddableWithArrayPersistentAttribute<T> {
+        @Id
+        int id;
+
+        FlattenedEmbeddableNestedWithArrayPersistentAttribute<T> v;
+    }
+
+    @Entity
+    static class ItemWithNestedEmbeddableWithCollectionPersistentAttribute<T> {
+        @Id
+        int id;
+
+        FlattenedEmbeddableNestedWithCollectionPersistentAttribute<T> v;
+    }
+
+    @Entity
+    static class ItemWithAggregateEmbeddableWithBasicPersistentAttribute<T> {
+        @Id
+        int id;
+
+        AggregateEmbeddableWithBasicPersistentAttribute<T> v;
+    }
+
+    @Entity
+    static class ItemWithAggregateEmbeddableWithArrayPersistentAttribute<T> {
+        @Id
+        int id;
+
+        AggregateEmbeddableWithArrayPersistentAttribute<T> v;
+    }
+
+    @Entity
+    static class ItemWithAggregateEmbeddableWithCollectionPersistentAttribute<T> {
+        @Id
+        int id;
+
+        AggregateEmbeddableWithCollectionPersistentAttribute<T> v;
+    }
+
+    @Entity
+    static class ItemWithCollectionOfAggregateEmbeddableWithBasicPersistentAttribute<T> {
+        @Id
+        int id;
+
+        Collection<AggregateEmbeddableWithBasicPersistentAttribute<T>> v;
+    }
```

Comment:
If we have this entity, then we also should have `ItemWithArrayOfAggregateEmbeddableWithBasicPersistentAttribute`.

Reply:
Left open until we deal with https://github.com/mongodb/mongo-hibernate/pull/125#discussion_r2452801848.